lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1273,17 +1257,17 @@ void TypeChecker::computeDefaultAccessibility(ExtensionDecl *ED) {
     auto getTypeAccess = [this, ED](const TypeLoc &TL) -> Accessibility {
       if (!TL.getType())
         return Accessibility::Public;
-      auto scopeDC =
+      auto accessScope =
           TypeAccessScopeChecker::getAccessScope(TL.getType(),
                                                  ED->getDeclContext(),
                                                  TypeAccessScopeCache);
       // This is an error case and will be diagnosed elsewhere.
-      if (!scopeDC.hasValue())
+      if (!accessScope)
         return Accessibility::Public;
 
-      if (!scopeDC.getValue())
+      if (accessScope->isPublic())
         return Accessibility::Public;
-      if (isa<ModuleDecl>(scopeDC.getValue()))
+      if (isa<ModuleDecl>(accessScope->getDeclContext()))
```

Comment:
Starting to feel like this should become a query on AccessScope as well (`isInternal` or something). But that can happen later.